awx/main/models/schedules.py

```diff
@@ -35,6 +35,61 @@
 UTC_TIMEZONES = {x: tzutc() for x in dateutil.parser.parserinfo().UTCZONE}
 
 
+def _check_valid_tzid(rrules):
+    for r in rrules:
+        if r._dtstart and r._dtstart.tzinfo is None:
+            raise ValueError('A valid TZID must be provided (e.g., America/New_York)')
```

Comment:
Since this is not doing `if any(): raise` but has an explicit top-level loop, I think it's important to mention what actually is wrong and where.

Could you add the specific rrule details into the message? And maybe spell out the abbreviations? As in, change “TZID” to “time zone identifier”, for example.

But also, perhaps it'd be better to go through all the rules and then add details on the batch of broken rules?

I'm thinking something along the lines of
```suggestion
def _assert_timezone_id_is_valid(rrules) -> None:
    broken_rrules = [
        rrule for rrule in rrules
        if r._dtstart and r._dtstart.tzinfo is None
    ]
    if not broken_rrules:
        return
    raise ValueError(
        'The following rrules do not a timezone set: '
        f'{broken_rrules}. Please, set it for each '
        '(e.g., America/New_York).',
    ) from None
```